File: bad-tests/loop_construct_vector_clause.cpp
#include "acc_testsuite.h"
#include <vector>

#ifndef T1
/*T1: loop construct vector clause,V:2.0-2.7*/
int test1() {
    const int size = 100;
    int err = 0;
    std::vector<int> host_array(size), device_array(size);
    for (int i = 0; i < size; ++i) {
        host_array[i] = i;
        device_array[i] = 0;
    }

    #pragma acc data copyin(host_array[0:size]) copyout(device_array[0:size])
    {
        #pragma acc parallel loop vector
        for (int i = 0; i < size; ++i) {
            device_array[i] = host_array[i] * i;
        }
    }

    for (int i = 0; i < size; ++i) {
        if (device_array[i] != host_array[i] * i) {
            ++err;
            break;
        }
    }

    return err;
}
#endif

int main() {
    int failcode = 0;
    int failed;
#ifndef T1
    failed = 0;
    for (int x = 0; x < NUM_TEST_CALLS; ++x) {
        failed = failed + test1();
    }
    if (failed != 0) {
        failcode = failcode + (1 << 0);
    }
#endif
    return failcode;
}

File: gpt-3.5-ft-large/template-context-full/loop_construct_vector_clause.f90
#ifndef T1
!T1:loop construct vector clause,V:2.7-2.3
      LOGICAL FUNCTION test1()
        USE OPENACC
        IMPLICIT NONE
        INCLUDE "acc_testsuite.Fh"
        
        INTEGER :: errors = 0
        REAL(8),DIMENSION(10*P): a, b, c !Data length is 10

        SEEDDIM(1) = 1
         
        #pragma acc data copy(c(0:10*P)) copyin(a(0:10*P), b(0:10*P))
          #pragma acc kernels loop
          DO i = 1, 10 * P
            a(i) = GetRandomNumber(1)
            #pragma acc loop vector
            DO j = 1, i
              b(i) = GetRandomNumber(1)
              b(j) = GetRandomNumber(1)
              temp = (4 * a(i) + 2 * b(i))/i
              #pragma acc loop vector
              DO s = 1, i
                temp = temp - a(s)
              END DO
              c(i) = temp
            END DO
          END DO

        DO i = 0, 10*P
          data_test(a(i), b(i), c(i), i)
        END DO

        IF (errors .eq. 0) THEN
          test1 = .FALSE.
        ELSE
          test1 = .TRUE.
        END IF
      END
#endif

      PROGRAM main
        IMPLICIT NONE
        INTEGER :: failcode, testrun
        LOGICAL :: failed
        INCLUDE "acc_testsuite.Fh"
#ifndef T1
        LOGICAL :: test1
#endif
        failed = .FALSE.
        failcode = 0
#ifndef T1
        DO testrun = 1, NUM_TEST_CALLS
          failed = failed .or. test1()
        END DO
        IF (failed) THEN
          failcode = failcode + 2 ** 0
          failed = .FALSE.
        END IF
#endif
        CALL EXIT (failcode)
      END PROGRAM